clients/rospy/src/rospy/core.py

```diff
@@ -143,78 +145,107 @@ def rospywarn(msg, *args):
     """Internal rospy client library warn logging"""
     _rospy_logger.warn(msg, *args)
     
-logdebug = logging.getLogger('rosout').debug
 
-logwarn = logging.getLogger('rosout').warning
+def _base_logger(msg, *args, **kwargs):
+
+    try:
+        name = kwargs.pop('logger_name')
+    except KeyError:
+        name = None
+
+    try:
+        throttle = kwargs.pop('logger_throttle')
+    except KeyError:
+        throttle = None
+
+    try:
+        level = kwargs.pop('logger_level')
+    except KeyError:
+        level = None

```

Comment:
Looking back through my testing, I had a test case where using the named arguments for name, throttle and level (like above) would interfer with variable length arguments.  This happened when testing out code that was using the existing public API, the named variables would end up taking some of the values from the multi length arguments.